platform/ios/test/MGLMapViewLayoutTests.m

```diff
@@ -0,0 +1,119 @@
+#import <XCTest/XCTest.h>
+#import "MGLMapView.h"
+#import "MGLMapViewDelegate.h"
+#import "MGLAccountManager.h"
+
+
+@interface MGLMapViewLayoutTests : XCTestCase<MGLMapViewDelegate>
+
+@property (nonatomic) UIView *superView;
+@property (nonatomic) MGLMapView *mapView;
+@property (nonatomic) XCTestExpectation *styleLoadingExpectation;
+
+@end
+
+@implementation MGLMapViewLayoutTests
+
+- (void)setUp {
+    [super setUp];
+
+    [MGLAccountManager setAccessToken:@"pk.feedcafedeadbeefbadebede"];
+    NSURL *styleURL = [[NSBundle bundleForClass:[self class]] URLForResource:@"one-liner" withExtension:@"json"];
+    NSLog(@"================> Screen size: %@", NSStringFromCGRect(UIScreen.mainScreen.bounds));
+
+    self.superView = [[UIView alloc] initWithFrame:UIScreen.mainScreen.bounds];
+
+    self.mapView = [[MGLMapView alloc] initWithFrame:UIScreen.mainScreen.bounds styleURL:styleURL];
+    self.mapView.delegate = self;
+
+    [self.superView addSubview:self.mapView];
+
+    UIView *mapView = self.mapView;
+    NSDictionary *bindings = NSDictionaryOfVariableBindings(mapView);
+    NSArray *verticalConstraints = [NSLayoutConstraint constraintsWithVisualFormat:@"V:|-0-[mapView]-0-|" options:0 metrics:nil views:bindings];
+    NSArray *horizonatalConstraints = [NSLayoutConstraint constraintsWithVisualFormat:@"H:|-0-[mapView]-0-|" options:0 metrics:nil views:bindings];
+
+    [self.superView addConstraints:[verticalConstraints arrayByAddingObjectsFromArray:horizonatalConstraints]];
+
+    self.styleLoadingExpectation = [self expectationWithDescription:@"Map view should finish loading style."];
+    [self waitForExpectationsWithTimeout:1 handler:nil];
+
+    //set zoom and heading so that scale bar and compass will be shown
+    [self.mapView setZoomLevel:4.5 animated:NO];
+    [self.mapView.camera setHeading:12.0];
+
+    //invoke layout
+    [self.superView setNeedsLayout];
+    [self.superView layoutIfNeeded];
+    NSLog(@"================> self.mapView.frame: %@", NSStringFromCGRect(self.mapView.frame));
+    if ( [self.mapView respondsToSelector:@selector(safeAreaInsets)] ) {
+        NSLog(@"================> self.mapView.safeAreaInsets: %@", NSStringFromUIEdgeInsets(self.mapView.safeAreaInsets));
+    }
+}
+
+- (void)mapView:(MGLMapView *)mapView didFinishLoadingStyle:(MGLStyle *)style {
+    XCTAssertNotNil(mapView.style);
+    XCTAssertEqual(mapView.style, style);
+
+    [self.styleLoadingExpectation fulfill];
+}
+
+- (void)tearDown {
+    self.styleLoadingExpectation = nil;
+    self.mapView = nil;
+
+    [super tearDown];
+}
+
+// TODO: check constraining to superview vs. constraining to safeAreaLayoutGuide.bottomAnchor
```

Comment:
I think you may have addressed this TODO? I thought we had an issue with layout depending on how the superview was constraining the map view. Please either delete this comment or test both cases (happy to pair on this if it's not clear)